mesa/space.py

```diff
@@ -197,8 +197,12 @@ def neighbor_iter(
             moore: Boolean for whether to use Moore neighborhood (including
                    diagonals) or Von Neumann (only up/down/left/right).
         """
-        neighborhood = self.get_neighborhood(pos, moore=moore)
-        return self.iter_cell_list_contents(neighborhood)
+        from warnings import warn
+        warn(
+            "`neighbor_iter` is deprecated in favor of `iter_neighbors` "
+            "and will be removed in the subsequent version."
+        )
```

Comment:
Notice I intentionally didn't use `DeprecationWarning`, because when I tested this warn, it didn't show up when I used `DeprecationWarning`. It will show up as `UserWarning`.